pilot/pkg/bootstrap/server.go

```diff
@@ -289,8 +292,11 @@ func (s *Server) initMonitor(args *PilotArgs) error {
 }
 
 func (s *Server) initClusterRegistries(args *PilotArgs) (err error) {
-	if args.Config.ClusterRegistriesDir != "" {
-		s.clusterStore, err = clusterregistry.ReadClusters(args.Config.ClusterRegistriesDir)
+	if args.Config.ClusterRegistriesConfigmap != "" {
+		s.clusterStore, err = clusterregistry.ReadClusters(s.kubeClient,
```

Comment:
If the plan is to watch (or maybe use a CRD and the existing watchers for CRD): sounds ok.

As it is I don't see the difference compared with loading from the dir, as before - the old code was just simpler.

Probably not a big issue - since eventually we want to be able to add cluster dynamically, and that implies watch.